backend/src/main/java/de/echochamber/backend/controller/PollController.java

```diff
@@ -50,26 +52,83 @@ public PollController(PollService pollsService, UserService userService) {
         return ok(map(createdPollEntity));
     }
 
-    private Poll map(PollEntity pollEntity){
+    @PostMapping(value = "/answer/{answerId}")
+    public ResponseEntity<Answer> giveAnswer(
+            @AuthenticationPrincipal UserEntity authUser,
+            @PathVariable Long answerId) {
+        PossibleAnswerEntity possibleAnswerEntity = answerService.findPossibleAnswerById(answerId);
+        Optional<UserEntity> userEntityOptional = userService.findByUserName(authUser.getUserName());
+
+        if (userEntityOptional.isPresent()) {
+
+            AnswerEntity givenAnswer = AnswerEntity.builder()
+                    .chosenAnswer(possibleAnswerEntity)
+                    .user(userEntityOptional.get())
+                    .build();
+            return ok(map(pollsService.giveAnswer(givenAnswer)));
+        }
+        throw new EntityNotFoundException("User not found");
+    }
+
+    @GetMapping
+    public ResponseEntity<List<Poll>> findAll() {
+        List<PollEntity> pollEntities = pollsService.findAll();
+        return ok(map(pollEntities));
+    }
+
+    private List<Poll> map(List<PollEntity> pollEntities) {
+        List<Poll> polls = new ArrayList<>();
+        for (PollEntity pollEntity : pollEntities) {
+            polls.add(map(pollEntity));
+        }
+        return polls;
+    }
+
+    private Answer map(AnswerEntity answerEntity) {
+        return Answer.builder()
+                .answer(answerEntity.getChosenAnswer().getAnswer())
+                .user(map(answerEntity.getUser())).build();
+    }
+
+    private User map(UserEntity userEntity) {
+        return User.builder()
+                .userName(userEntity.getUserName())
+                .firstName(userEntity.getFirstName())
+                .lastName(userEntity.getLastName())
+                .avatarUrl(userEntity.getAvatarUrl()).build();
+    }
+
+    private Poll map(PollEntity pollEntity) {
         User user = User.builder()
                 .userName(pollEntity.getCreatedBy().getUserName())
                 .firstName(pollEntity.getCreatedBy().getFirstName())
                 .lastName(pollEntity.getCreatedBy().getLastName())
                 .role(pollEntity.getCreatedBy().getRole())
                 .avatarUrl(pollEntity.getCreatedBy().getAvatarUrl()).build();
 
-        List<String> possibleAnswerEntities = pollEntity.getPossibleAnswers().stream().map(answer-> answer.getAnswer()).toList();
-
+        List<String> possibleAnswerEntities = pollEntity.getPossibleAnswers().stream().map(answer -> answer.getAnswer()).toList();
```

Comment:
das ist ein bißchen versteckt, vlt im Controller das übergeben und das maping übernimmt nur das maping? EIgentlich ist es nicht so wichtig, aber ich musste mich teils echt anstrengen um das nachzuvollziehen ;) , nicht dass dir dasselbe in 3-4 Wochen auch passiert